adr:Create a Harbor robot account (#439)
elastisys/compliantkubernetes: docs/adr/0024-create-Harbor-robot-account.md

> # Create a Harbor robot account that cab create other robot accounts with full privileges

Reviewer: ```suggestion
# Create a Harbor robot account that can create other robot accounts with full privileges
```

Author: fixed

> # Create a Harbor robot account that can create other robot accounts with full privileges

Reviewer 2: ```suggestion
# Allow a Harbor robot account that can create other robot accounts with full privileges
```

Author: Thx
Fixed

* Status: accepted
* Deciders: arch meeting

> * Date: 2022-03-03

Reviewer 2: ```suggestion
* Date: 2022-11-17
```

Author: Hmmm
I wonder how/why I put that date....
Thank you for noticing it :)

## Context and Problem Statement

We offer UI access to Harbor with admin privileges.

> Customer uses a Harbor operator that needs an admiin robot account with privileges to create other robot accounts with full prvileges.

Reviewer: ```suggestion
Customer uses a Harbor operator that needs an admin robot account with privileges to create other robot accounts with full privileges.
```

Author: fixed

> 

Reviewer 2: Let's add a problem statement formulated as a question:

```
Should we allow creation of a Harbor robot account which can create other Harbor robot accounts?

Author: Added

## Decision Drivers

> * We want to deliver a easy to use and automate platform.

Reviewer: ```suggestion
* We want to deliver a platform that is easy to use and easy to automate.
```

Author: fixed

> * We want to deliver a platform that is easy to use and easy to automate.

Reviewer 2: * We want to ensure platform security and stability.
* We want to make it hard for customers to break the platform via trivial mistakes.

Author: Thank you

## Considered Options

> * Do not offer this

Reviewer 2: ```suggestion
* Do not allow Harbor robot account which can create Harbor robot accounts
```

Author: Looks and sounds better
Thx

> * Offer this as it does not provide additional privileges, but instead offer a more flexible platform.

Reviewer 2: ```suggestion
* Allow Harbor robot accounts which can create Harbor robot accounts
```

Author: Thx

## Decision Outcome

> Chosen option: "Offer this as it does not provide additional privileges"

Reviewer 2: ```suggestion
Chosen option: "Allow Harbor robot accounts which can create Harbor robot accounts", because it does not provide additional privileges, but instead offers a more self-service platform.
```

### Positive Consequences

* Harbor is more flexible and easy to automate

### Negative Consequences

> * Increases the chances that the customer can criple the Harbor service.

Reviewer: ```suggestion
* Increases the chances that the customer can cripple the Harbor service.
```

Author: fixed

## Recommendations to Operators

> Make it clear in the ticket requesting this that the customer accepts the risc of "shooting themselves in the foot"

Reviewer: ```suggestion
Make it clear in the ticket requesting this that the customer accepts the risk of "shooting themselves in the foot"
```

Author: fixed
